packages/grid/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -610,7 +614,15 @@ const GridCellV7 = React.forwardRef<HTMLDivElement, GridCellV7Props>((props, ref
   const cellRef = React.useRef<HTMLDivElement>(null);
   const handleRef = useForkRef(ref, cellRef);
   const focusElementRef = React.useRef<FocusElement>(null);
-  const ownerState = { align, showRightBorder, isEditable, classes: rootProps.classes, isSelected };
+  const isSelectionMode = rootProps.unstable_cellSelection;
```

Comment:
Nit: To keep the type (boolean) consistent incase of MIT use-cases:
```suggestion
  const isSelectionMode = rootProps.unstable_cellSelection ?? false;
```